# Importamos librerias  	


In [2]:
import pandas as pd
import numpy as np

pd.set_option("display.max_columns", None)

# Importamos BB.DD

In [3]:
df_orig = pd.read_csv("datos/netflix_originals.csv", index_col=0)
df_titles= pd.read_csv("datos/netflix_titles.csv",index_col=0)

# 2. Analisis de columnas y consistencias

## Netflix_Titles - df_titles

### Visualizacion general del df

In [4]:
df_titles

show_id     type                  title         director  \
0         s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1         s2  TV Show          Blood & Water              NaN   
2         s3  TV Show              Ganglands  Julien Leclercq   
3         s4  TV Show  Jailbirds New Orleans              NaN   
4         s5  TV Show           Kota Factory              NaN   
...      ...      ...                    ...              ...   
8802   s8803    Movie                 Zodiac    David Fincher   
8803   s8804  TV Show            Zombie Dumb              NaN   
8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805   s8806    Movie                   Zoom     Peter Hewitt   
8806   s8807    Movie                 Zubaan      Mozez Singh   

                                                   cast        country  \
0                                                   NaN  United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2     Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                   NaN            NaN   
4     Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   
...                                                 ...            ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8803                                                NaN            NaN   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...          India   

              date_added  release_year rating   duration  \
0     September 25, 2021          2020  PG-13     90 min   
1     September 24, 2021          2021  TV-MA  2 Seasons   
2     September 24, 2021          2021  TV-MA        NaN   
3     September 24, 2021          2021  TV-MA        NaN   
4     September 24, 2021          2021  TV-MA  2 Seasons   
...                  ...           ...    ...        ...   
8802   November 20, 2019          2007      R    158 min   
8803        July 1, 2019          2018  TV-Y7  2 Seasons   
8804    November 1, 2019          2009      R        NaN   
8805    January 11, 2020          2006     PG        NaN   
8806       March 2, 2019          2015  TV-14    111 min   

                                              listed_in  \
0                                         Documentaries   
1       International TV Shows, TV Dramas, TV Mysteries   
2     Crime TV Shows, International TV Shows, TV Act...   
3                                Docuseries, Reality TV   
4     International TV Shows, Romantic TV Shows, TV ...   
...                                                 ...   
8802                     Cult Movies, Dramas, Thrillers   
8803             Kids' TV, Korean TV Shows, TV Comedies   
8804                            Comedies, Horror Movies   
8805                 Children & Family Movies, Comedies   
8806     Dramas, International Movies, Music & Musicals   

                                            description  
0     As her father nears the end of his life, filmm...  
1     After crossing paths at a party, a Cape Town t...  
2     To protect his family from a powerful drug lor...  
3     Feuds, flirtations and toilet talk go down amo...  
4     In a city of coaching centers known to train I...  
...                                                 ...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...  

[8807 rows x 12 columns]

### Visualizacion info especifica del df

In [5]:
df_titles.info()

<class 'pandas.core.frame.DataFrame'>
Index: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      3994 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 894.5+ KB


- Con la funcion info podemos observar que exites 8807 peliculas, series y documentales. 
- La mayoria de las columnas son de tipo objeto, menos el año de lanzamiento. Lo curioso es que la duracion esta puesta como objeto(string), al igual que la fecha de lanzamiento. 
- Tenemos muchos datos nulos en lo que es directores y ni hablar a lo que se refiere en la duracion.

### Hacemos un analisis descriptivo de los titulos

In [6]:
df_titles.describe(include=['object', 'category'])

show_id   type   title       director                cast  \
count     8807   8807    8807           6173                7982   
unique    8807      2    8807           4528                7692   
top      s8807  Movie  Zubaan  Rajiv Chilaka  David Attenborough   
freq         1   6131       1             19                  19   

              country       date_added rating   duration  \
count            7976             8797   8803       3994   
unique            748             1767     17        185   
top     United States  January 1, 2020  TV-MA  2 Seasons   
freq             2818              109   3207        425   

                           listed_in  \
count                           8807   
unique                           514   
top     Dramas, International Movies   
freq                             362   

                                              description  
count                                                8807  
unique                                               8775  
top     Paranormal activity at a lush, abandoned prope...  
freq                                                    4

- podemos ver que existen unicamente 2 tipos de contenidos -> peliculas y series
- titulos todos diferentes por lo que no hay valores duplicados
- directores hay 4528 que tienen su contenido en plataforma
- Respecto a los paises el que mayor contenido produce es estados unidos con 2818. 
- el genero con mayor contenido es lo que son los dramas internacionales.

In [7]:
df_titles.describe().T

count         mean       std     min     25%     50%     75%  \
release_year  8807.0  2014.180198  8.819312  1925.0  2013.0  2017.0  2019.0   

                 max  
release_year  2021.0

Respecto a variable numerica a analizar solo contamos con el año de lanzamiento
- La media dio en 2014 pero la mediana es 2017
- El 75% del contenido se lanzo antes de 2019.

### Checkeamos nulos de df_titles

- ### Creamos funcion buscar_nulos() para poder pasar dataframe que quiseramos y asi encontrar aquellos valores faltantes en cada columna

In [8]:
def buscar_nulos(df):
    #Creamos Df vacio
    df_titulos_nulos=pd.DataFrame()

    df_titulos_nulos["Numero_Nulos"] = df.isnull().sum() #Creamos columna de nulos -> si isnull()=True= 1 => si sumamos todos, nos da el numero de nulos
    df_titulos_nulos["porcentaje_nulos"] = round((df.isnull().sum()/df.shape[0])*100,2) #Para saber el porcenaje de nulos, lo dividimos sobre la cantidad de filas que tiene nuesto df
    return df_titulos_nulos.T



In [9]:
buscar_nulos(df_titles) 	

show_id  type  title  director    cast  country  date_added  \
Numero_Nulos          0.0   0.0    0.0   2634.00  825.00   831.00       10.00   
porcentaje_nulos      0.0   0.0    0.0     29.91    9.37     9.44        0.11   

                  release_year  rating  duration  listed_in  description  
Numero_Nulos               0.0    4.00   4813.00        0.0          0.0  
porcentaje_nulos           0.0    0.05     54.65        0.0          0.0

- Casi el 30% de la categoria director no tenemos conocimiento de quien hizo ese contenido
- Otro dato grave que desconocemos es el la duracion del contenido. Mas del 54% desconocemos cuanto dura.
- Con valores similares, el reparto y de que pais proviene el contenido, tienen un poco mas del 9% de nulos. 

### Checkeamos duplicados de df_titles

#### Creamos funcion para reconocer duplicados

In [10]:
def duplicados(df):
    return df.duplicated().sum()

### Buscando en todo el data frame de titulos.

In [11]:
duplicados(df_titles)

np.int64(0)

- No contamos con valores duplicados en la base de datos de titulos de netflix.

### Buscando especificamente si hay titulos repetidos

In [12]:
duplicados(df_titles["title"])

np.int64(0)

- Tampoco contamos con duplicados en lo que es especificamente los titulos. Es decir, no tenemos ningun titulo que se repite de los 8807.

## Creamos funcion para ver los valores unicos por columna

In [13]:
def valores_unicos_columnas(df, columna=None):
    #creamos df vacio
    df_val_unico=pd.DataFrame()
    # llenamos el df con los tiposs y valores
    df_val_unico["tipo"] = df[columna].unique()
    df_val_unico["cantidad"] = df[columna].value_counts().values

    return df_val_unico

## Vemos que tipo de contenido es el mayor subido

In [14]:
valores_unicos_columnas(df_titles, columna="type")

tipo  cantidad
0    Movie      6131
1  TV Show      2676

- la mayoria del contenido de la bbdd de titulos son peliculas

# Conclusiones EDA -> df_titles
- Existen columnas de datos que no tienen el tipo correcto de dato. Por ejemplo: Duration(hay algunos que dicen 2 seasons) y fecha de agregacion (esta en formato largo) a la plataforma
- Tomar accion con los nulos en df_titles
    -   Por del 54% no tenemos conocimiento de cuanto dura el contenido. Altamente grave ya que no podemos saber que prefiere la gente cuando se trata de duracion.
    -   Casi el 30% del contenido no tiene directores. 
    -   Casi el 10% del contenido no tiene el reparto ni de que pais es el mismo.
    -   Podemos omitir el contenido date_added y rating rellenandolo con la mediana de cada uno, o la moda. 
- No contamos con datos duplicados en la bbdd ya que todos los titulos son diferentes
- Existen unicamente 2 tipos de contenidos -> peliculas y series
- Contamos con informacion de que hay 4528 directores en la plataforma
- Respecto a los paises, el que mayor contenido produce es EEUU con 2818. 
- El genero con mayor contenido es lo que son los dramas internacionales.


## Netflix_originals - df_orig


### Visualizacion general del df

In [15]:
df_orig

Title                  Genre  \
0                                Enter the Anime            Documentary   
1                                    Dark Forces               Thriller   
2                                        The App  Science fiction/Drama   
3                                 The Open House        Horror thriller   
4                                    Kaali Khuhi                Mystery   
..                                           ...                    ...   
576                       Cuba and the Cameraman            Documentary   
577                       Dancing with the Birds            Documentary   
580  Winter on Fire: Ukraine's Fight for Freedom            Documentary   
581                      Springsteen on Broadway           One-man show   
583     David Attenborough: A Life on Our Planet            Documentary   

              Premiere  Runtime  IMDB Score                  Language  
0       August 5, 2019       58         2.5          English/Japanese  
1      August 21, 2020       81         2.6                   Spanish  
2    December 26, 2019       79         2.6                   Italian  
3     January 19, 2018       94         3.2                   English  
4     October 30, 2020       90         3.4                     Hindi  
..                 ...      ...         ...                       ...  
576  November 24, 2017      114         8.3                   English  
577   October 23, 2019       51         8.3                   English  
580    October 9, 2015       91         8.4  English/Ukranian/Russian  
581  December 16, 2018      153         8.5                   English  
583    October 4, 2020       83         9.0                   English  

[513 rows x 6 columns]

### Visualizacion info especifica del df

In [16]:
df_orig.info()

<class 'pandas.core.frame.DataFrame'>
Index: 513 entries, 0 to 583
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Title       513 non-null    object 
 1   Genre       513 non-null    object 
 2   Premiere    513 non-null    object 
 3   Runtime     513 non-null    int64  
 4   IMDB Score  513 non-null    float64
 5   Language    513 non-null    object 
dtypes: float64(1), int64(1), object(4)
memory usage: 28.1+ KB


Respecto a los titulos originales de netflix podemos detacar que:
- No se encontraron nulos
- Son un total de 513 peliculas, series y documentales
- Tenemos la duracion de todos los titulos que en el df posterior no teniamos de muchos.

### Hacemos un analisis descriptivo de los titulos

In [17]:
df_orig.describe(include=['object', 'category']).T

count unique                                       top freq
Title      513    513  David Attenborough: A Life on Our Planet    1
Genre      513    106                               Documentary  132
Premiere   513    350                          October 18, 2019    5
Language   513     37                                   English  352

- Destacamos lo relevante que son los documentales en los contenidos exclusivos de netflix 
- Del total de 513 elementos, 352 son en ingles. 
- El dia con mayores lanzamientos en la plataforma fue el 18/10/2019

In [18]:
df_orig.describe().T

count       mean        std  min   25%   50%    75%    max
Runtime     513.0  94.674464  26.648121  4.0  87.0  97.0  108.0  209.0
IMDB Score  513.0   6.210916   0.968850  2.5   5.6   6.3    6.9    9.0

Respecto a las variables numericas:
- La nota de IMDB: 
    - La media del contenido de netflix es de 6,21 estrellas, esto muy similar a la mediana de 6,3. 
    - Casi el 25% del contenido de netflix tiene una nota superior a 7 estrellas, por lo que el contenido que produce netflix es muy mediocre.
- La duracion del contenido:
    - Media y mediana muy similares en tiempo, entre 94 y 97 minutos respectivamente
    - El 75% de las peliculas no llegan a las 2horas. Es decir en su mayoria son cortas.


### Chequeamos nulos con funcion declarada previamente

In [19]:
buscar_nulos(df_orig)

Title  Genre  Premiere  Runtime  IMDB Score  Language
Numero_Nulos        0.0    0.0       0.0      0.0         0.0       0.0
porcentaje_nulos    0.0    0.0       0.0      0.0         0.0       0.0

Como vimos en el cuadro anterior, no existian valores nulos.

### Chequeamos duplicados con funcion declarada previamente

In [20]:
duplicados(df_orig)

np.int64(0)

No existen valores duplicados.

In [21]:
valores_unicos_columnas(df_orig, columna="Genre")


tipo  cantidad
0                             Documentary       132
1                                Thriller        73
2                   Science fiction/Drama        42
3                         Horror thriller        35
4                                 Mystery        33
..                                    ...       ...
101                      Action-adventure         1
102                          Concert Film         1
103           Animation / Science Fiction         1
104                 Anthology/Dark comedy         1
105  Animation/Christmas/Comedy/Adventure         1

[106 rows x 2 columns]

# Conclusiones EDA -> df_orig

- 513 elementos originales de netflix
- Los documentales de netflix son los mas populares, seguidos por los thrillersm 
- Del total de 513 elementos, 352 son en idioma de ingles. 
- El dia con mayores lanzamientos en la plataforma fue el 18/10/2019
- No se encontraron valores duplicados ni nulos.
Respecto a las variables numericas:
- La media del contenido de netflix es de 6,21 estrellas, esto muy similar a la mediana de 6,3. 
- Casi el 25% del contenido de netflix tiene una nota superior a 7 estrellas, por lo que el contenido que produce netflix es muy mediocre.
- La duracion Media y mediana muy similares en tiempo, entre 94 y 97 minutos respectivamente
- El 75% de las peliculas no llegan a las 2horas. Es decir en su mayoria son cortas.

# PARTE 2: Union de datos

### Creamos columna en df_orig para distinga previo a unificacion, cuales son originales de netflix

In [22]:
df_orig["Es_original"] = 1
df_orig.sample(1)

Title            Genre            Premiere  Runtime  IMDB Score  \
331  Gerald's Game  Horror thriller  September 29, 2017      103         6.5   

    Language  Es_original  
331  English            1

### Cambiamos el nombre de la columna para que sea el mismo que el de df_titles

In [23]:
df_orig.rename(columns={'Title': 'title'}, inplace=True)
df_orig.sample(3)

title            Genre        Premiere  Runtime  IMDB Score  \
310     Take Your Pills      Documentary  March 16, 2018       87         6.4   
256       Someone Great  Romantic comedy  April 19, 2019       92         6.2   
45   Mrs. Serial Killer         Thriller     May 1, 2020      106         4.8   

    Language  Es_original  
310  English            1  
256  English            1  
45     Hindi            1

### Ahora si podemos hacer mergeo entre df_titles y df_orig por medio de la columna "title"

In [24]:
df_contenido = pd.merge(df_titles, df_orig, on="title", how="left")
df_contenido.set_index('show_id', inplace=True)
df_contenido.sample(10)

type                                title  \
show_id                                                 
s8571      Movie                      Theater of Life   
s3037    TV Show         Jamtara - Sabka Number Ayega   
s4455      Movie                Oye Lucky! Lucky Oye!   
s4672      Movie                             Sparring   
s2493      Movie  The Witness Who Didn't See Anything   
s4135      Movie                          Manusangada   
s4364    TV Show                Minecraft: Story Mode   
s3726    TV Show                              Shooter   
s3503    TV Show                    Criminal: Germany   
s3314    TV Show                  The Devil Next Door   

                            director  \
show_id                                
s8571                   Peter Svatek   
s3037                Soumendra Padhi   
s4455               Dibakar Banerjee   
s4672                    Samuel Jouy   
s2493    Muhammad Fadl, Hany Motawie   
s4135                   Amshan Kumar   
s4364                            NaN   
s3726                            NaN   
s3503                            NaN   
s3314                            NaN   

                                                      cast  \
show_id                                                      
s8571                                      Massimo Bottura   
s3037    Amit Sial, Dibyendu Bhattacharya, Aksha Pardha...   
s4455    Abhay Deol, Paresh Rawal, Neetu Chandra, Archa...   
s4672    Mathieu Kassovitz, Olivia Merilahti, Souleyman...   
s2493    Adel Imam, Nahed Gabr, Omar El-Hariri, Nazim S...   
s4135    Rajeev Anand, Manimegalai, A.S. Sasi Kumar, Vi...   
s4364    Patton Oswalt, Catherine Taber, Sean Astin, Me...   
s3726    Ryan Phillippe, Omar Epps, David Marciano, Cyn...   
s3503    Eva Meckbach, Sylvester Groth, Florence Kasumb...   
s3314                                                  NaN   

                       country          date_added  release_year rating  \
show_id                                                                   
s8571                   Canada       July 18, 2017          2016  TV-PG   
s3037                    India    January 10, 2020          2020  TV-MA   
s4455                    India    November 1, 2018          2008  TV-14   
s4672                   France   September 1, 2018          2017  TV-MA   
s2493                    Egypt        May 21, 2020          1976  TV-PG   
s4135                    India    February 2, 2019          2017  TV-14   
s4364    United States, Sweden   November 27, 2018          2015  TV-PG   
s3726            United States       June 21, 2019          2018  TV-14   
s3503                  Germany  September 20, 2019          2019  TV-MA   
s3314            United States    November 4, 2019          2019  TV-MA   

        duration                                          listed_in  \
show_id                                                               
s8571        NaN                                      Documentaries   
s3037        NaN  Crime TV Shows, International TV Shows, TV Dramas   
s4455    118 min             Comedies, Dramas, International Movies   
s4672        NaN        Dramas, International Movies, Sports Movies   
s2493    157 min                     Comedies, International Movies   
s4135        NaN   Dramas, Independent Movies, International Movies   
s4364        NaN                                           Kids' TV   
s3726        NaN   Crime TV Shows, TV Action & Adventure, TV Dramas   
s3503        NaN  British TV Shows, Crime TV Shows, Internationa...   
s3314        NaN                         Crime TV Shows, Docuseries   

                                               description Genre Premiere  \
show_id                                                                     
s8571    This documentary follows chef Massimo Bottura ...   NaN      NaN   
s3037    A group of small-town young men run a lucrativ...   NaN      NaN   
s4455    Starting as a petty thief, a m

Se realizo este tipo de mergeo para integrar la información adicional contenida en df_orig con el conjunto de datos principal df_titles. La columna key utilizada para la combinación es "title", que representa el vínculo lógico entre ambos DataFrames.
Se utilizó un left join para garantizar que todas las filas del DataFrame principal df_titles se mantengan en el resultado final, incluso si no hay una coincidencia en df_orig. Esto asegura que no se pierda ninguna información de los títulos presentes en df_titles,mientras que las filas que no tengan datos coincidentes en df_orig simplemente se rellenarán con NaN.

Que podemos ver a un primer vistazo como punto negativo en el mergeo:
- Tenemos muchos valores nulos con respecto a las categorias que no contabamos en df_titles. Seria muy interesante saber: IMDB, el genero y el lenguaje

# PARTE 2: Uso de `loc` e `iloc`

1. **Filtrado de datos con `loc`:**

- Selecciona todas las filas donde el tipo de contenido sea “Movie”.

In [25]:
df_contenido.loc[df_contenido.type=="Movie"]

type                             title  \
show_id                                            
s1       Movie              Dick Johnson Is Dead   
s7       Movie  My Little Pony: A New Generation   
s8       Movie                           Sankofa   
s10      Movie                      The Starling   
s13      Movie                      Je Suis Karl   
...        ...                               ...   
s8802    Movie                           Zinzana   
s8803    Movie                            Zodiac   
s8805    Movie                        Zombieland   
s8806    Movie                              Zoom   
s8807    Movie                            Zubaan   

                              director  \
show_id                                  
s1                     Kirsten Johnson   
s7       Robert Cullen, José Luis Ucha   
s8                        Haile Gerima   
s10                     Theodore Melfi   
s13                Christian Schwochow   
...                                ...   
s8802                  Majid Al Ansari   
s8803                    David Fincher   
s8805                  Ruben Fleischer   
s8806                     Peter Hewitt   
s8807                      Mozez Singh   

                                                      cast  \
show_id                                                      
s1                                                     NaN   
s7       Vanessa Hudgens, Kimiko Glenn, James Marsden, ...   
s8       Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...   
s10      Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...   
s13      Luna Wedler, Jannis Niewöhner, Milan Peschel, ...   
...                                                    ...   
s8802    Ali Suliman, Saleh Bakri, Yasa, Ali Al-Jabri, ...   
s8803    Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...   
s8805    Jesse Eisenberg, Woody Harrelson, Emma Stone, ...   
s8806    Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...   
s8807    Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...   

                                                   country  \
show_id                                                      
s1                                           United States   
s7                                                     NaN   
s8       United States, Ghana, Burkina Faso, United Kin...   
s10                                          United States   
s13                                Germany, Czech Republic   
...                                                    ...   
s8802                         United Arab Emirates, Jordan   
s8803                                        United States   
s8805                                        United States   
s8806                                        United States   
s8807                                                India   

                 date_added  release_year rating duration  \
show_id                                                     
s1       September 25, 2021          2020  PG-13   90 min   
s7       September 24, 2021          2021     PG      NaN   
s8       September 24, 2021          1993  TV-MA  125 min   
s10      September 24, 2021          2021  PG-13  104 min   
s13      September 23, 2021          2021  TV-MA  127 min   
...                     ...           ...    ...      ...   
s8802         March 9, 2016          2015  TV-MA      NaN   
s8803     November 20, 2019          2007      R  158 min   
s8805      November 1, 2019          2009      R      NaN   
s8806      January 11, 2020          2006     PG      NaN   
s8807         March 2, 2019          2015  TV-14  111 min   

                                                listed_in  \
show_id                                                     
s1                                          Documentaries   
s7                               Children & Family Movies   
s8       Dramas, Independent Movies, International Movies   
s10                                      Comedies, Dramas   
s13         

- Muestra solo los títulos (title) y la duración (duration) de todos los contenidos que sean de tipo “TV Show”.

In [26]:
tv_show = df_contenido.loc[df_contenido.type=="TV Show"]
tv_show[["title","duration"]]

title   duration
show_id                                  
s2               Blood & Water  2 Seasons
s3                   Ganglands        NaN
s4       Jailbirds New Orleans        NaN
s5                Kota Factory  2 Seasons
s6               Midnight Mass        NaN
...                        ...        ...
s8796          Yu-Gi-Oh! Arc-V  2 Seasons
s8797               Yunus Emre  2 Seasons
s8798                Zak Storm        NaN
s8801       Zindagi Gulzar Hai        NaN
s8804              Zombie Dumb  2 Seasons

[2676 rows x 2 columns]

- Selecciona todas las filas donde el país sea “United States”.

In [27]:
df_contenido.loc[df_contenido.country=="United States"]

type                 title         director  \
show_id                                                   
s1         Movie  Dick Johnson Is Dead  Kirsten Johnson   
s10        Movie          The Starling   Theodore Melfi   
s16      TV Show     Dear White People              NaN   
s28        Movie             Grown Ups     Dennis Dugan   
s29        Movie            Dark Skies    Scott Stewart   
...          ...                   ...              ...   
s8792      Movie           Young Adult    Jason Reitman   
s8794      Movie  Yours, Mine and Ours     Raja Gosnell   
s8803      Movie                Zodiac    David Fincher   
s8805      Movie            Zombieland  Ruben Fleischer   
s8806      Movie                  Zoom     Peter Hewitt   

                                                      cast        country  \
show_id                                                                     
s1                                                     NaN  United States   
s10      Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...  United States   
s16      Logan Browning, Brandon P. Bell, DeRon Horton,...  United States   
s28      Adam Sandler, Kevin James, Chris Rock, David S...  United States   
s29      Keri Russell, Josh Hamilton, J.K. Simmons, Dak...  United States   
...                                                    ...            ...   
s8792    Charlize Theron, Patton Oswalt, Patrick Wilson...  United States   
s8794    Dennis Quaid, Rene Russo, Sean Faris, Katija P...  United States   
s8803    Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
s8805    Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
s8806    Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   

                 date_added  release_year rating duration  \
show_id                                                     
s1       September 25, 2021          2020  PG-13   90 min   
s10      September 24, 2021          2021  PG-13  104 min   
s16      September 22, 2021          2021  TV-MA      NaN   
s28      September 20, 2021          2010  PG-13      NaN   
s29      September 19, 2021          2013  PG-13      NaN   
...                     ...           ...    ...      ...   
s8792     November 20, 2019          2011      R      NaN   
s8794     November 20, 2019          2005     PG      NaN   
s8803     November 20, 2019          2007      R  158 min   
s8805      November 1, 2019          2009      R      NaN   
s8806      January 11, 2020          2006     PG      NaN   

                                    listed_in  \
show_id                                         
s1                              Documentaries   
s10                          Comedies, Dramas   
s16                    TV Comedies, TV Dramas   
s28                                  Comedies   
s29           Horror Movies, Sci-Fi & Fantasy   
...                                       ...   
s8792    Comedies, Dramas, Independent Movies   
s8794      Children & Family Movies, Comedies   
s8803          Cult Movies, Dramas, Thrillers   
s8805                 Comedies, Horror Movies   
s8806      Children & Family Movies, Comedies   

                                               description        Genre  \
show_id                                                                   
s1       As her father nears the end of his life, filmm...  Documentary   
s10      A woman adjusting to life after a loss contend...          NaN   
s16      Students of color navigate the daily slights a...          NaN   
s28      Mourning the loss of their beloved junior high...          NaN   
s29      A family’s idyllic suburban life shatters when...          NaN   
...                                                    ...          ...   
s8792    When a divorced writer gets a letter from an o...          NaN   
s8794    When a father of eight and a mother of 10 prep...          NaN   
s8803    A political cartoonist, a crime reporter and a...  

- Muestra todas las películas (type = “Movie”) y selecciona las columnas title y director.

In [39]:
movie = df_contenido.loc[df_contenido.type=="Movie"]
tv_show[["title","director"]]

title         director
show_id                                        
s2               Blood & Water              NaN
s3                   Ganglands  Julien Leclercq
s4       Jailbirds New Orleans              NaN
s5                Kota Factory              NaN
s6               Midnight Mass    Mike Flanagan
...                        ...              ...
s8796          Yu-Gi-Oh! Arc-V              NaN
s8797               Yunus Emre              NaN
s8798                Zak Storm              NaN
s8801       Zindagi Gulzar Hai              NaN
s8804              Zombie Dumb              NaN

[2676 rows x 2 columns]

- Selecciona los títulos (title) y géneros (listed_in) de todas las producciones lanzadas en 2018.

In [29]:
df2018=df_contenido.loc[df_contenido.release_year==2018]
df2018[["title","listed_in"]]

title  \
show_id                                                      
s22                                 Resurrection: Ertugrul   
s38                                            Angry Birds   
s74                                           King of Boys   
s95                                              Show Dogs   
s97      If I Leave Here Tomorrow: A Film About Lynyrd ...   
...                                                    ...   
s8733                                        White Chamber   
s8744                                             Wildlife   
s8775                                            يوم الدين   
s8776                                      Yeh Meri Family   
s8804                                          Zombie Dumb   

                                                 listed_in  
show_id                                                     
s22      International TV Shows, TV Action & Adventure,...  
s38                                  Kids' TV, TV Comedies  
s74                           Dramas, International Movies  
s95                     Children & Family Movies, Comedies  
s97                        Documentaries, Music & Musicals  
...                                                    ...  
s8733                          Sci-Fi & Fantasy, Thrillers  
s8744                           Dramas, Independent Movies  
s8775     Dramas, Independent Movies, International Movies  
s8776                  International TV Shows, TV Comedies  
s8804               Kids' TV, Korean TV Shows, TV Comedies  

[1147 rows x 2 columns]

- Selecciona las filas donde la columna director esté vacía y muestra solo los títulos (title).

In [30]:
direct_0 = df_contenido.loc[df_contenido.director.isna()]
direct_0[["title"]]

title
show_id                                     
s2                             Blood & Water
s4                     Jailbirds New Orleans
s5                              Kota Factory
s11      Vendetta: Truth, Lies and The Mafia
s15          Crime Stories: India Detectives
...                                      ...
s8796                        Yu-Gi-Oh! Arc-V
s8797                             Yunus Emre
s8798                              Zak Storm
s8801                     Zindagi Gulzar Hai
s8804                            Zombie Dumb

[2634 rows x 1 columns]

2. **Acceso a datos específicos con `iloc`:**

In [31]:
df_contenido.sample()

type                         title        director  \
show_id                                                        
s7134    Movie  Jis Desh Men Ganga Behti Hai  Radhu Karmakar   

                                                      cast country  \
show_id                                                              
s7134    Raj Kapoor, Padmini, Pran, Chanchal, Lalita Pa...     NaN   

                date_added  release_year rating duration  \
show_id                                                    
s7134    December 31, 2019          1960  TV-14  181 min   

                                                listed_in  \
show_id                                                     
s7134    Action & Adventure, Dramas, International Movies   

                                               description Genre Premiere  \
show_id                                                                     
s7134    Falling in with a band of outlaws during a hol...   NaN      NaN   

         Runtime  IMDB Score Language  Es_original  
show_id                                             
s7134        NaN         NaN      NaN          NaN

   - Selecciona las primeras 5 filas del conjunto de datos y muestra solo las columnas title, director y country.

In [32]:
df5 = df_contenido.iloc[0:5,[1,2,4]]
df5

title         director        country
show_id                                                       
s1        Dick Johnson Is Dead  Kirsten Johnson  United States
s2               Blood & Water              NaN   South Africa
s3                   Ganglands  Julien Leclercq            NaN
s4       Jailbirds New Orleans              NaN            NaN
s5                Kota Factory              NaN          India

   - Muestra las últimas 5 filas del conjunto de datos y selecciona las columnas title y description.


In [33]:
df6 = df_contenido.iloc[-5:,[1,10]]
df6

title                                        description
show_id                                                                
s8803         Zodiac  A political cartoonist, a crime reporter and a...
s8804    Zombie Dumb  While living alone in a spooky town, a young g...
s8805     Zombieland  Looking to survive in a world taken over by zo...
s8806           Zoom  Dragged from civilian life, a former superhero...
s8807         Zubaan  A scrappy but poor boy worms his way into a ty...

- Selecciona las primeras 10 filas del dataset y muestra las columnas desde la segunda hasta la quinta (incluidas). 

In [34]:
df10 = df_contenido.iloc[0:10,1:6]
df10

title                       director  \
show_id                                                                    
s1                   Dick Johnson Is Dead                Kirsten Johnson   
s2                          Blood & Water                            NaN   
s3                              Ganglands                Julien Leclercq   
s4                  Jailbirds New Orleans                            NaN   
s5                           Kota Factory                            NaN   
s6                          Midnight Mass                  Mike Flanagan   
s7       My Little Pony: A New Generation  Robert Cullen, José Luis Ucha   
s8                                Sankofa                   Haile Gerima   
s9          The Great British Baking Show                Andy Devonshire   
s10                          The Starling                 Theodore Melfi   

                                                      cast  \
show_id                                                      
s1                                                     NaN   
s2       Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   
s3       Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...   
s4                                                     NaN   
s5       Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...   
s6       Kate Siegel, Zach Gilford, Hamish Linklater, H...   
s7       Vanessa Hudgens, Kimiko Glenn, James Marsden, ...   
s8       Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...   
s9       Mel Giedroyc, Sue Perkins, Mary Berry, Paul Ho...   
s10      Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...   

                                                   country          date_added  
show_id                                                                         
s1                                           United States  September 25, 2021  
s2                                            South Africa  September 24, 2021  
s3                                                     NaN  September 24, 2021  
s4                                                     NaN  September 24, 2021  
s5                                                   India  September 24, 2021  
s6                                                     NaN  September 24, 2021  
s7                                                     NaN  September 24, 2021  
s8       United States, Ghana, Burkina Faso, United Kin...  September 24, 2021  
s9                                          United Kingdom  September 24, 2021  
s10                                          United States  September 24, 2021

   - Muestra las últimas 7 filas del dataset y selecciona las columnas title, director, y country usando índices posicionales.

In [35]:
df_7 = df_contenido.iloc[-7:,[1,2,4]]
df_7

title         director                       country
show_id                                                                   
s8801    Zindagi Gulzar Hai              NaN                      Pakistan
s8802               Zinzana  Majid Al Ansari  United Arab Emirates, Jordan
s8803                Zodiac    David Fincher                 United States
s8804           Zombie Dumb              NaN                           NaN
s8805            Zombieland  Ruben Fleischer                 United States
s8806                  Zoom     Peter Hewitt                 United States
s8807                Zubaan      Mozez Singh                         India

   - Selecciona las filas desde la 15 hasta la 25 (incluidas) y muestra las primeras 4 columnas.

In [36]:
df15 = df_contenido.iloc[15:26,0:4]
df15

type                                              title  \
show_id                                                               
s16      TV Show                                  Dear White People   
s17        Movie  Europe's Most Dangerous Man: Otto Skorzeny in ...   
s18      TV Show                                    Falsa identidad   
s19        Movie                                          Intrusion   
s20      TV Show                                             Jaguar   
s21      TV Show    Monsters Inside: The 24 Faces of Billy Milligan   
s22      TV Show                             Resurrection: Ertugrul   
s23        Movie                                    Avvai Shanmughi   
s24        Movie       Go! Go! Cory Carson: Chrissy Takes the Wheel   
s25        Movie                                              Jeans   
s26      TV Show                               Love on the Spectrum   

                                              director  \
show_id                                                  
s16                                                NaN   
s17      Pedro de Echave García, Pablo Azorín Williams   
s18                                                NaN   
s19                                         Adam Salky   
s20                                                NaN   
s21                                    Olivier Megaton   
s22                                                NaN   
s23                                     K.S. Ravikumar   
s24                            Alex Woo, Stanley Moore   
s25                                         S. Shankar   
s26                                                NaN   

                                                      cast  
show_id                                                     
s16      Logan Browning, Brandon P. Bell, DeRon Horton,...  
s17                                                    NaN  
s18      Luis Ernesto Franco, Camila Sodi, Sergio Goyri...  
s19      Freida Pinto, Logan Marshall-Green, Robert Joh...  
s20      Blanca Suárez, Iván Marcos, Óscar Casas, Adriá...  
s21                                                    NaN  
s22      Engin Altan Düzyatan, Serdar Gökhan, Hülya Dar...  
s23      Kamal Hassan, Meena, Gemini Ganesan, Heera Raj...  
s24      Maisie Benson, Paul Killam, Kerry Gudjohnsen, ...  
s25      Prashanth, Aishwarya Rai Bachchan, Sri Lakshmi...  
s26                                       Brooke Satchwell

- Muestra las filas 20 a 30 y solo las columnas title, release_year, y rating usando índices posicionales.

In [37]:
df2030 = df_contenido.iloc[20:31,[1,6,7]]
df2030

title  release_year rating
show_id                                                                      
s21      Monsters Inside: The 24 Faces of Billy Milligan          2021  TV-14
s22                               Resurrection: Ertugrul          2018  TV-14
s23                                      Avvai Shanmughi          1996  TV-PG
s24         Go! Go! Cory Carson: Chrissy Takes the Wheel          2021   TV-Y
s25                                                Jeans          1998  TV-14
s26                                 Love on the Spectrum          2021  TV-14
s27                                       Minsara Kanavu          1997  TV-PG
s28                                            Grown Ups          2010  PG-13
s29                                           Dark Skies          2013  PG-13
s30                                             Paranoia          2013  PG-13
s31                                      Ankahi Kahaniya          2021  TV-14

# GUARDAMOS EL CONJUNTO DE DATOS

In [38]:
df_contenido.to_csv("df_contenido.csv", index=True)  